src/saltext/vmware/utils/license_mgr.py

```diff
@@ -0,0 +1,399 @@
+    else:
+        if datacenter_name:
+            # need to get named datacenter's reference
+            datacenter_ref = utils_datacenter.get_datacenter(service_instance, datacenter_name)
+            log.debug(
+                f"retrieved datacenter ref '{datacenter_ref }' for datacenter '{datacenter_name}'"
+            )
+
+        if cluster_name:
+            # need to get named cluster's reference
+            cluster_ref = utils_cluster.get_cluster(dc_ref=datacenter_ref, cluster=cluster_name)
+            _entity_id = cluster_ref._moId
+            log.debug(
+                f"retrieved entityId '{_entity_id}' from cluster ref '{cluster_ref }' for cluster '{cluster_name}' and datacenter '{datacenter_name}'"
+            )
+
+        if esxi_hostname:
+            # need to get named esxi server
+            esxi_hosts = utils_esxi.get_hosts(service_instance, datacenter_name, esxi_hostname)
+
+            # returns hosts list of dicts
+            if not esxi_hosts:
+                log.debug(f"Failed to find esxi hostname '{esxi_hostname}'")
+                return False
+
+            if len(esxi_hosts) > 1:
+                log.error(
+                    f"Failed, found multiple instances of esxi hostname '{esxi_hostname}', hosts returned '{esxi_hosts}'"
+                )
+                return False
+
+            _entity_id = esxi_hosts[0]._moId
+
+    return _entity_id
+
+
+def is_vcenter(service_instance):
+    """
+    Test if service_instance represents vCenter,
+
+    service_instance
+        The Service Instance to check if it represents a vCenter
+
+    Return:
+        True  - vCenter
+        False - not a vCenter
+    """
+    try:
+        srv_content = get_service_content(service_instance)
+        apitype = srv_content.about.apiType
+    except vim.fault.VimFault as exc:
+        log.exception(exc)
+        raise salt.exceptions.VMwareApiError(exc.msg)
+    except vmodl.RuntimeFault as exc:
+        log.exception(exc)
+        raise salt.exceptions.VMwareRuntimeError(exc.msg)
+    except AttributeError as exc:
+        log.exception(exc)
+        raise salt.exceptions.CommandExecutionError(exc.msg)
+    except TypeError as exc:
+        log.exception(exc)
+        raise salt.exceptions.CommandExecutionError(exc.msg)
+
+    if apitype == "VirtualCenter":
+        return True
+
+    return False
+
+
+def get_license_mgr(service_instance):
+    """
+    Return a license manager from specified Service Instance
+    if the Service Instance is connected to a vCenter,
+    otherwise return None.
+
+    service_instance
+        vCenter service connection instance
+
+    Note: returns LicenseManager
+    """
+    if not is_vcenter(service_instance):
+        return None
+
+    srv_content = get_service_content(service_instance)
+    return srv_content.licenseManager
+
+
+def get_license_assignment_mgr(service_instance):
+    """
+    Return a license assignment manager from specified Service Instance
```

Comment:
**suggestion** since the `any` bit needs to be changed, may as well update this one: `Return a license assignment manager from specified Service Instance if the Service Instance is connected to a vCenter, otherwise return None`